lua/neotest-mocha/init.lua

```diff
@@ -175,12 +175,11 @@ function Adapter.build_spec(args)
   local pos = tree:data()
   local testNamePattern = "'.*'"
 
-  if pos.type == "test" then
-    testNamePattern = "'" .. util.escape_test_pattern(pos.name) .. "$'"
-  end
-
-  if pos.type == "namespace" then
-    testNamePattern = "'^" .. util.escape_test_pattern(pos.name) .. "'"
+  if pos.type == "test" or pos.type == "namespace" then
+    local testName = string.sub(pos.id, string.find(pos.id, "::") + 2)
+    testName, _ = string.gsub(testName, "::", " ")
+    testNamePattern = "'^" .. util.escape_test_pattern(testName)
```

Comment:
This logic has changed to always include a `'^'` since we want to match the specific full position id (without filename).